src/modules/AutoinstSoftware.rb

```diff
@@ -1092,15 +1081,13 @@ def ReadHelper
     #    "packages" -> list<string> user selected packages
     #           "remove-packages" -> list<string> packages to remove
     def read_initial_stage
-      install_patterns = Pkg.ResolvableProperties("", :pattern, "").collect do |pattern|
-        # Do not take care about if the pattern has been selected by the user or the product
-        # definition, cause we need a base selection here for the future
-        # autoyast installation. (bnc#882886)
-        if pattern["user_visible"] &&
-            (pattern["status"] == :selected || pattern["status"] == :installed)
-          pattern["name"]
+      install_patterns =
+        Y2Packager::Resolvable.find(kind: :pattern, user_visible: true).collect do |pattern|
```

Comment:
Please, prefer `#map` to `#collect` (is basically an alias, but we usually use `#map`).